app/src/androidTest/java/com/github/sdp_begreen/begreen/activities/DatabaseActivityTest.kt

```diff
@@ -8,40 +8,36 @@ import androidx.test.espresso.matcher.ViewMatchers.*
 import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import androidx.test.filters.LargeTest
-import com.github.sdp_begreen.begreen.Database
+import com.github.sdp_begreen.begreen.FirebaseDB
 import com.github.sdp_begreen.begreen.R
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import org.junit.After
 import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
-import java.util.concurrent.CompletableFuture
+
 
 @RunWith(AndroidJUnit4::class)
 @LargeTest
 class DatabaseActivityTest {
-    @get:Rule
-    val activityRule = ActivityScenarioRule(DatabaseActivity::class.java)
-
-    /*
-    Implementation of a local database
-     */
-    class MockDataBase : Database() {
-        private val map: HashMap<String, String> = HashMap()
-        override fun get(key: String): CompletableFuture<String> {
-            return CompletableFuture.completedFuture(map[key])
-        }
 
-        override fun set(key: String, value: String) {
-            map[key] = value
-        }
+    @Before
+    fun init() {
+        Firebase.database.goOffline()
```

Comment:
Here I think it would be better to use the firebase emulator to test our database implementation, it will avoid polluting production database each time we tests the app.

```suggestion
        @Before
            fun init() {
                Firebase.database.useEmulator("10.0.2.2",9000)
            }
```

Reply:
Indeed, the Emulator seemed to be painful to set up but it looks like it wasnt that much, thanks a lot!